src/main/groovy/com/bmuschko/gradle/docker/tasks/image/Dockerfile.groovy

```diff
@@ -501,20 +501,31 @@ class Dockerfile extends DefaultTask {
     }
 
     interface ItemJoiner {
-        String join(Map map)
+        String join(Map<String, String> map)
     }
 
     static class MultiItemJoiner implements ItemJoiner {
         @Override
-        String join(Map map) {
-            map.inject([]) { result, entry -> result << "\"$entry.key\"=\"$entry.value\"" }.join(' ')
+        String join(Map<String, String> map) {
+            map.inject([]) { result, entry ->
+                // TODO: clarify if assert should be used even in productive code and if this would be a valid one either
+                assert !entry.key.matches('.*(?: |(?:\r{0,1}\n)).*')
+                def key = !entry.key.matches('["].*["]') && entry.key.contains(' ') ? "$entry.key" : entry.key
+                def value = !entry.value.matches('["].*["]') && entry.value.matches('.*(?: |(?:\r{0,1}\n)).*') ? '"'.concat(entry.value.replaceAll('"', '\\\\"')).concat('"') : entry.value
+                value = value.replaceAll("(\r)*\n", "\\\\\n")
+                result << "$key=$value"
+            }.join(' ')
         }
     }
 
     static class SingleItemJoiner implements ItemJoiner {
         @Override
-        String join(Map map) {
-            map.inject([]) { result, entry -> result << "$entry.key $entry.value" }.join(' ')
+        String join(Map<String, String> map) {
+            map.inject([]) { result, entry ->
+                assert !entry.key.matches('.*(?: |(?:\r{0,1}\n)).*')
```

Comment:
Lets throw an exception here with an appropriate error message thereby letting the user know what exactly popped and why.